### ISSUE TITLE
Qgis server 3.10.3 issues

### ISSUE THREAD

**Issue by loic74** [qgis/QGIS#35107]
With qgis server 3.10.3 and 3.10.3 projects (on Ubuntu bionic), Qgis server crashes after the first request :

`
2020/03/16 11:39:04 [error] 1277#1277: *33 upstream prematurely closed connection while reading response header from upstream, client: 192.168.0.11, server: 192.168.0.11, request: "GET /qgisserver/qgis_mapserv.fcgi?map=%2Fhome%2Fsig%2Fwebmapping%2Ddata%2Fprojets%2Feaux%2Feau%5Fassainissement%2Eqgs&request=GetCapabilities&service=WMS&version=1%2E3%2E0 HTTP/1.1", upstream: "fastcgi://unix:/var/run/qgisserver.socket:", host: "192.168.0.11"
`

`
2020/03/16 11:39:04 [error] 1277#1277: *36 connect() to unix:/var/run/qgisserver.socket failed (111: Connection refused) while connecting to upstream, client: 192.168.0.11, server: 192.168.0.11, request: "GET /qgisserver/qgis_mapserv.fcgi?map=%2Fhome%2Fsig%2Fwebmapping%2Ddata%2Fprojets%2Feaux%2Feau%5Fassainissement%2Eqgs&request=GetCapabilities&service=WMS HTTP/1.1", upstream: "fastcgi://unix:/var/run/qgisserver.socket:", host: "192.168.0.11"
`

Qgis server 3.10.3 does not crash with 3.4.13 projects. But, in this case, getprint shows the wrong layers if any group layer is specified.

Thanks for your help.

**Comment by rldhont** [comment]
About crashes, how did you configure QGIS Server ?
* apache with mod_fcgi
* nginx with spawn-fcgi
* nginx with fcgiwrap
 
what are the environment variables you have set ?

**Comment by loic74** [comment]
nginx with spawn-fcgi

**Comment by jgrocha** [comment]
@loic74 Do you have QGIS Server log enabled? Can you check the output of QGIS Server log? Maybe with the log enabled, we can see where the QQGIS Server crashes.

I usually set:
```
QGIS_SERVER_LOG_FILE=/var/log/qgis/qgisserver.log
QGIS_SERVER_LOG_LEVEL=1
```

Check the available log options from https://docs.qgis.org/testing/en/docs/user_manual/working_with_ogc/server/config.html

**Comment by gisgogo** [comment]
At qgis-server version 3.12 same problem with print as 3.10.

**Comment by jgrocha** [comment]
> At qgis-server version 3.12 same problem with print as 3.10.

@gisgogo Can you provide more details? It would be great if you provide the QGIS log. It would be perfect if you provide a small project that fails to print. For example, you can create a project with a few layers and store it in a GeoPackage, including the project itself.

**Comment by gisgogo** [comment]
Hi. Qgis 3.10 and 3.12, qgis-server 3.10 and 3.12 (3.4 was ok). Project is here: https://chudenice.gisland.eu/index.php/view/map/?repository=tech&project=katastr_nemovitosti-covid

Not print layer "budovy", log is here:
10:47:07 INFO Server[25678]: ******************** New request ***************
10:47:07 INFO Server[25678]: Request URL: https://map.gisland.eu:443/cgi-bin/qgis_mapserv.fcgi?
10:47:07 INFO Server[25678]: Environment:
10:47:07 INFO Server[25678]: ------------------------------------------------
10:47:07 INFO Server[25678]: SERVER_NAME: map.gisland.eu
10:47:07 INFO Server[25678]: REQUEST_URI: /cgi-bin/qgis_mapserv.fcgi
10:47:07 INFO Server[25678]: SCRIPT_NAME: /cgi-bin/qgis_mapserv.fcgi
10:47:07 INFO Server[25678]: HTTPS: on
10:47:07 INFO Server[25678]: REMOTE_ADDR: 127.0.0.1
10:47:07 INFO Server[25678]: SERVER_PORT: 443
10:47:07 INFO Server[25678]: QUERY_STRING: 
10:47:07 INFO Server[25678]: CONTENT_TYPE: application/x-www-form-urlencoded
10:47:07 INFO Server[25678]: REQUEST_METHOD: POST
10:47:07 INFO Server[25678]: HTTP_ACCEPT: */*
10:47:07 INFO Server[25678]: HTTP_USER_AGENT: Lizmap
10:47:07 WARNING [25678]: Warning: error parsing post data as XML: at line 1, column 1: error occurred while parsing element. Assuming urlencoded query string sent in the post body.
10:47:07 WARNING [25678]: formatString is: pdf
10:47:07 INFO wfsOutputExtension[25678]: WFSFilter.requestReady
10:47:07 INFO Server[25678]: DPI:100
10:47:07 INFO Server[25678]: EXCEPTIONS:application/vnd.ogc.se_inimage
10:47:07 INFO Server[25678]: FORMAT:pdf
10:47:07 INFO Server[25678]: LAYERS:budovy,Katastrální mapa,čísla popisná
10:47:07 INFO Server[25678]: MAP:/var/www/lizmap/lizmap-web-client-3.3.5/lizmap/install/chudenice-tech/katastr_nemovitosti-covid.qgs
10:47:07 INFO Server[25678]: MAP0:EXTENT:1467417.925534,6353643.790941,1467672.931666,6353818.79515
10:47:07 INFO Server[25678]: MAP0:LAYERS:budovy,Katastrální mapa,čísla popisná
10:47:07 INFO Server[25678]: MAP0:SCALE:1000
10:47:07 INFO Server[25678]: MAP0:STYLES:,,
10:47:07 INFO Server[25678]: OPACITIES:255,255,255
10:47:07 INFO Server[25678]: REQUEST:GetPrint
10:47:07 INFO Server[25678]: SERVICE:WMS
10:47:07 INFO Server[25678]: SRS:EPSG:3857
10:47:07 INFO Server[25678]: STYLES:,,
10:47:07 INFO Server[25678]: TEMPLATE:chudenice
10:47:07 INFO Server[25678]: TRANSPARENT:true
10:47:07 INFO Server[25678]: VERSION:1.3.0
10:47:07 INFO Server[25678]: WMS Request parameters:
10:47:07 INFO Server[25678]:  - SRS : EPSG:3857
10:47:07 INFO Server[25678]:  - LAYERS : budovy,Katastrální mapa,čísla popisná
10:47:07 INFO Server[25678]:  - OPACITIES : 255,255,255
10:47:07 INFO Server[25678]:  - FORMAT : pdf
10:47:07 INFO Server[25678]:  - TRANSPARENT : true
10:47:07 INFO Server[25678]:  - DPI : 100
10:47:07 INFO Server[25678]:  - TEMPLATE : chudenice
10:47:07 INFO Server[25678]:  - VERSION : 1.3.0
10:47:07 INFO wfsOutputExtension[25678]: WFSFilter.responseComplete
10:47:07 INFO Server[25678]: Request finished in 250 ms

**Comment by gisgogo** [comment]
I think, problem is with layers at group make at Lizmap. At Lizmap git write, problem is at qgis-server viz.  https://github.com/3liz/lizmap-web-client/issues/1555

**Comment by rldhont** [comment]
@gisgogo which layer is a group in budovy, Katastrální mapa, čísla popisná ?

**Comment by gisgogo** [comment]
Here is my project. [https://cloud.gisland.eu/index.php/s/XW7FXR9qaEcQ9gN](url)

**Comment by rldhont** [comment]
Issue confirmed in test #35277

**Comment by rldhont** [comment]
I confirm the regression, the test created passes on branch release-3_4 and fails on master.

**Comment by rldhont** [comment]
To be sure, has the QGIS project been saved with QGIS 3.10 or upper ?

**Comment by gisgogo** [comment]
Yes, project was saved at qgis 3.10 and 3.12.

**Comment by rldhont** [comment]
First PR https://github.com/qgis/QGIS/pull/35277

Second PR https://github.com/qgis/QGIS/pull/35937

Last backport https://github.com/qgis/QGIS/pull/35949

**Comment by rldhont** [comment]
@loic74 fix about WMS GetPrint regression #37425 has been merged in release-3_10 and will be available in 3.10.8.

Can we close this issue ?

**Comment by loic74** [comment]
@rldhont Yes, of course. Thanks for resolving this issue ;-)
